src/http/server.cr

```diff
@@ -159,6 +162,9 @@ class HTTP::Server
 
   # Adds a `Socket::Server` *socket* to this server.
   def bind(socket : Socket::Server) : Nil
+    raise "Can't add socket to running server." if listening?
+    raise "Can't add socket to closed server." if @wants_close
```

Comment:
Just disallow `#close` `unless listening?`. I'd also just rename this variable to `@closed` because the state persists after it's fully shut down - not just when it's in some "closing down" state that "wants close" implies.

Reply:
You need to be able to call `#close` even if the server is not listening because sockets may already be bound and need to be released.

Reply:
having `getter? closed` would be nice too.

Reply:
Already in there ;)